File: fm-domain/src/main/java/io/zeitwert/fm/util/CrnkUtils.java
package io.zeitwert.fm.util;

import io.crnk.core.queryspec.FilterSpec;
import io.crnk.core.queryspec.QuerySpec;

public class CrnkUtils {

	public static boolean hasFilterFor(QuerySpec querySpec, String fieldName) {
		return querySpec.getFilters().stream().anyMatch(f -> getPath(f).equals(fieldName));
	}

	public static String getPath(FilterSpec filter) {
		return String.join(".", filter.getPath().getElements()).replace(".id", "Id");
	}

}

File: fm-domain/src/main/java/io/zeitwert/app/api/jsonapi/dto/CrnkUtils.kt
package io.zeitwert.app.api.jsonapi.dto

import io.crnk.core.queryspec.FilterSpec
import io.crnk.core.queryspec.QuerySpec

object CrnkUtils {

	fun hasFilterFor(
		querySpec: QuerySpec,
		fieldName: String,
	): Boolean = querySpec.filters.any { getPath(it) == fieldName }

	@JvmStatic
	fun getPath(filter: FilterSpec): String =
		filter.path.elements
			.joinToString(".")
			.replace(".id", "Id")

}
